File: Communication entre processus/exo3.2.py
import os
import sys

def main():
    if len(sys.argv) < 2:
        sys.exit(1)

    fd_lecture = int(sys.argv[1])
    
    tampon = os.read(fd_lecture, 100)
    print(f"Fils (execv) a reçu : {tampon.decode().strip()}")

    os.close(fd_lecture)

if __name__ == "__main__":
    main()

File: Communication entre processus/exo3.2.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

int main(int argc, char *argv[]) 
{
    if (argc < 2) 
    {
        return 1;
    }
    int fd_lecture = atoi(argv[1]); 
    char tampon[100];

    read(fd_lecture, tampon, 100);
    printf("Fils (execv) a reçu : %s\n", tampon);

    close(fd_lecture);
    return 0;
}